cedar-policy/src/api.rs

```diff
@@ -1809,6 +1864,70 @@ impl Policy {
     /// If `id` is Some, the policy will be given that Policy Id.
     /// If `id` is None, then "JSON policy" will be used.
     /// The behavior around None may change in the future.
+    ///
+    /// ```
+    /// use cedar_policy::{Policy, PolicyId};
+    /// use std::str::FromStr;
+    ///
+    /// let data = r#"
+    ///        {
+    ///            "effect":"permit",
+    ///            "principal":{
+    ///            "op":"==",
+    ///            "entity":{
+    ///                "type":"User",
+    ///                "id":"bob"
+    ///            }
+    ///            },
+    ///            "action":{
+    ///            "op":"==",
+    ///            "entity":{
+    ///                "type":"Action",
+    ///                "id":"view"
+    ///            }
+    ///            },
+    ///            "resource":{
+    ///            "op":"==",
+    ///            "entity":{
+    ///                "type":"Album",
+    ///                "id":"trip"
+    ///            }
+    ///            },
+    ///            "conditions":[
+    ///            {
+    ///                "kind":"when",
+    ///                "body":{
+    ///                   ">":{
+    ///                        "left":{
+    ///                        ".":{
+    ///                            "left":{
+    ///                                "Var":"principal"
+    ///                            },
+    ///                            "attr":"age"
+    ///                        }
+    ///                        },
+    ///                        "right":{
+    ///                        "Value":18
+    ///                        }
+    ///                    }
+    ///                }
+    ///            }
+    ///            ]
+    ///        }
+    /// "#;
+    /// let v = serde_json::from_str(data).unwrap();
+    /// let policy = Policy::from_json(None, v).unwrap();
+    /// let src = r#"
+    /// permit(
+    ///   principal == User::"bob",
+    ///   action == Action::"view",
+    ///   resource == Album::"trip"
+    /// )
+    /// when
+    ///   { principal.age > 18 };"#;
+    /// let expected_output = Policy::parse(None, src).unwrap();
+    /// assert_eq!(policy.to_string(), expected_output.to_string());
```

Comment:
should this be `assert_eq!(policy.to_string(), src);`? Why do we even need to call `Policy::parse()`?

Reply:
`assert_eq!(policy.to_string(), src);`  fails because `policy.to_string()` prints `principal.age > 18` in `src` as `18 < (principal[\"age\"])`.

Reply:
Ah, right.